formats/properties/commonMain/src/kotlinx/serialization/properties/Properties.kt

```diff
@@ -102,7 +103,7 @@ public sealed class Properties(
         }
 
         final override fun <T> decodeSerializableValue(deserializer: DeserializationStrategy<T>): T {
-            val type = map["type"]?.toString()
+            val type = map[nested("type")]?.toString()
```

Comment:
I would put this statement under the if condition below. Most types are not polymorphic, and here we will have an unnecessary creation of a nested key.

Reply:
Good idea, thank you, I changed it.